source/backend/discovery/src/lib/awsClient.js

```diff
@@ -553,7 +526,7 @@ function createCognitoClient(credentials, region) {
 
     // describeUserPool has an RPS of 15, we set this artificially low to avoid
     // being rate limited
-    const describeUserPoolThrottler = pThrottle({
+    const describeUserPoolThrottler = createThrottler('describeUserPool', credentials, region, {
```

Comment:
NIT: 'describe' naming conventions are alternating between 'describe' and resource type being the first word.

Reply:
Where possible I've just named them after the API they're invoking.

Reply:
If we're talking about `elbDescribe` and `ecsDescribe`, we need a name that means the memoization is disctint. The pattern is to go from more general to specific when required. E.g., <service-name>-<verb>-<resource> where we only use <service-name> when required but maybe we should make service name mandatory to ensure uniqueness?

Reply:
that sounds good to me!

Reply:
done!

Reply:
lgtm